### PR TITLE
Add test for get_city

### PR DESCRIPTION
**Pull request by ipwnponies**
This is a start of the test boilerplate. I'm thinking of adding the tests in separate PRs if it helps to keep discussions focused. Or a single big PR because it's just unit test tedium.

This resolves the bare minimum for #14 but I should have enough data to hit effective 100% coverage.

### CODE REVIEW

File: mealpy/mealpy.py
Comment (ipwnponies) [inline mealpy/mealpy.py:75]: This will raise error and cause script to fail. I guess we should consider handling that gracefully and hook it into the retry logic. That's out of scope for this PR though, it's existing behaviour.

Comment (edmundmok) [reply mealpy/mealpy.py:75]: Agree that we should handle the exceptions in a separate PR, and do this for other API methods too.

File: tests/mealpy_test.py
Comment (edmundmok) [inline tests/mealpy_test.py:23]: How about pulling out these mock responses as a test fixture? It looks like you are reusing them (at least partially) in `test_get_city_not_found`.

File: tests/mealpy_test.py
Comment (edmundmok) [inline tests/mealpy_test.py:73]: What does the `>=` do here? I have never used this way of asserting responses so it is quite unclear to me. Perhaps you can make use of the above mock response to do an equality check?

**Review by edmundmok — COMMENTED** [review]
Just some minor comments.

File: tests/mealpy_test.py
Comment (ipwnponies) [reply tests/mealpy_test.py:23]: I'm holding off on turning them to fixtures. Depending on how the other tests look, it might or might not end up being heavily reused.

I'm trying to strike a balance between documenting the full response one can expect (as test data for implementing new features) and over-specifying. Which is why `test_get_city_not_found` has a stripped down response because it doesn't care about all the other attributes.

File: tests/mealpy_test.py
Comment (ipwnponies) [reply tests/mealpy_test.py:73]: This is a new pattern I'm trying out, inspired from jest javascript testing patterns. It's asserting that `city.items` is a superset of the minimum object we need, which contains values for `id` and `name`.

[items()](https://docs.python.org/3/library/stdtypes.html#dict-views) returns a set-like object, `>=` is [`issuperset`](https://docs.python.org/3/library/stdtypes.html#frozenset.issuperset). Otherwise the expected value would need to be a much bigger and more specified object. This replaces the deprecated [`unittest.TestCase.assertDictContainsSubset`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertDictContainsSubset) function.

Comment (edmundmok) [reply tests/mealpy_test.py:73]: Interesting, the symbol makes more sense now.

**Review by edmundmok — APPROVED** [review]

File: tests/mealpy_test.py
Comment (ipwnponies) [reply tests/mealpy_test.py:23]: I should further add that from my experience, unit tests are easier to grok when they're dumb. Unlike systems, where composition is favoured for reuse, tests can start to have too much contextual overhead (it's a system of its own!) and then you can start introducing bugs in testing code. And who tests the tests...

I'm trying to preserve a full and accurate approximation of the "happy day" response, so it can be used as reference when developing and debuggin. But in edge case tests, I don't want to be weighed down by heavy weight objects.

**Comment by edmundmok** [comment]
BTW, feel free to merge as usual (though looks like there is a conflict currently) once approved!

**Pull request merged**
